sympy/matrices/tests/test_matrices.py

```diff
@@ -2378,6 +2378,11 @@ def test_issue_5320():
         [0, 1, 0, 2]
     ])
 
+def test_issue_11944():
+    A = Matrix([[1]])
+    AIm = sympify(A)
+    assert Matrix.hstack(AIm, A) == Matrix([[1, 1]])
+    assert Matrix.vstack(AIm, A) == Matrix([[1], [1]])
```

Comment:
It would be nice to add a test that cycles through every class in the `classes` variable at the top of the tests file and make sure the class is preserved.